.github/workflows/build_macos.yml

```diff
@@ -71,15 +67,11 @@ jobs:
             tmp_install/
           key: ${{ runner.os }}-pg-${{ steps.pg_ver.outputs.pg_rev }}
 
-      - name: Set extra env for macOS
-        if: matrix.os == 'macos-latest'
+      - name: Build Postgres header files
         run: |
           echo 'LDFLAGS=-L/usr/local/opt/openssl@3/lib' >> $GITHUB_ENV
           echo 'CPPFLAGS=-I/usr/local/opt/openssl@3/include' >> $GITHUB_ENV
-
-      - name: Build postgres
-        if: steps.cache_pg.outputs.cache-hit != 'true'
-        run: make postgres
+          make postgres-headers
```

Comment:
Would header building be able to check things like
https://github.com/neondatabase/postgres/pull/138
?

If it's not, let's do the whole build since there's no other linting for such erros.
Otherwise, we have to change the `Cache postgres build` and add `-01` or whatever prefix to invalidate the old, bigger one.